file-store/src/test/java/org/keycloak/models/map/storage/file/group/GroupDummyWriterTest.java

```diff
@@ -0,0 +1,256 @@
+            events.add(new ScalarEvent(Optional.empty(), Optional.empty(), implicitTuple, parentIdValue, ScalarStyle.PLAIN));
+        }
+
+        addEndEvents(events);
+
+        writeEventsTofile(events);
+    }
+
+    @Test
+    public void testALittleBitLessDummyWriteGroup() throws Exception {
+        String realmId = "realm1";
+
+        MapGroupEntity parentGroup = DeepCloner.DUMB_CLONER.newInstance(MapGroupEntity.class);
+        MapGroupEntity childGroup = DeepCloner.DUMB_CLONER.newInstance(MapGroupEntity.class);
+
+        parentGroup.setId("id1");
+        parentGroup.setName("parent group");
+        parentGroup.setRealmId(realmId);
+        parentGroup.setGrantedRoles(Set.of("role1", "role2", "role3"));
+        parentGroup.setAttribute("a0", List.of("v0"));
+        parentGroup.setAttribute("a1", List.of("v1, v2"));
+        parentGroup.setAttribute("a2", List.of("v3, v3, v4"));
+
+//        childGroup.setId("id2");
+//        childGroup.setName("child group");
+//        childGroup.setRealmId(realmId);
+//        childGroup.setParentId(parentGroup.getId());
+//        childGroup.setGrantedRoles(Set.of("role4"));
+
+        List<Event> events = new LinkedList<>();
+        addStartEvents(events);
+        addPairOfScalars(events, "schemaVersion", "1");
+
+        for (MapGroupEntityFields field : MapGroupEntityFields.values()) {
+            System.out.println("  adding field " + field.getName());
+            if (field.equals(MapGroupEntityFields.ID) || field.equals(MapGroupEntityFields.REALM_ID)) {
+                // do not write ID or REALM_ID into the file
+                // ID will be recorded in name of the file
+                // REALM_ID will be recorded by directory name
+                continue;
+            }
+            Object fieldVal = field.get(parentGroup);
+            if (fieldVal != null) { // field has some non-null value
+                // let's start with those used in MapGroupEntityFields only
+                if (field.getFieldClass().equals(String.class)) { // NAME, PARENT_ID this should be pair of scalars
+                    addPairOfScalars(events, field.getNameCamelCase(), fieldVal.toString());
+                } else if (field.getFieldClass().equals(Set.class)) { // GRANTED_ROLES
+                    String keyName = field.getNameCamelCase();
+                    Set<String> values = (Set) field.get(parentGroup); // for now we know it's Set<String> - TODO make it more general
+
+                    if (values.size() == 1) {
+                        addPairOfScalars(events, keyName, values.iterator().next());
+                    } else {
+                        addSequence(events, keyName, values);
+                    }
+                } else if (field.getFieldClass().equals(Map.class)) { // ATTRIBUTES
+                    addAttributes(events, parentGroup);
+                }
+            }
+        }
```

Comment:
The above is very interesting.. the code above is very generic, we can probably iterate through generic `EntityField` instead of the group-specific fields here.